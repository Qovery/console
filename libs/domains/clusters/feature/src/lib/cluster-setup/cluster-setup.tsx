import { Callout, ExternalLink, Icon } from '@qovery/shared/ui'
import CopyButton from './copy-button/copy-button'

export function ClusterSetup({ type }: { type: 'LOCAL_DEMO' | 'SELF_MANAGED' }) {
  return (
    <>
      {type === 'SELF_MANAGED' && (
        <Callout.Root color="sky">
          <Callout.Icon>
            <Icon iconName="info-circle" iconStyle="regular" />
          </Callout.Icon>
          <Callout.Text>
            <p>
              This guide walks you through the process of installing Qovery on your existing Kubernetes cluster, where
              you will have full control over the infrastructure. You will be responsible for managing updates and
              upgrades. Please note that advanced Kubernetes knowledge is required.
            </p>
          </Callout.Text>
        </Callout.Root>
      )}
      <ul className="flex flex-col gap-4 text-sm font-medium text-neutral-400">
        <li className="rounded border border-neutral-200 p-3">
          <h5 className="mb-1 text-sm font-medium">1. Download/Update Qovery CLI</h5>
          <p className="mb-2 font-normal text-neutral-350">
            Download and install the Qovery CLI (or update its version to the latest version).
          </p>
          <ExternalLink href="https://hub.qovery.com/docs/using-qovery/interface/cli/#install">
            https://hub.qovery.com/docs/using-qovery/interface/cli/#install
          </ExternalLink>
        </li>
        <li className="rounded border border-neutral-200 p-3">
          <h5 className="mb-1 text-sm font-medium">
            2. {type === 'LOCAL_DEMO' ? 'Install your cluster' : 'Install Qovery on your cluster'}
          </h5>
          <p className="mb-2  font-normal text-neutral-350">
            Run the following command from your terminal and follow the instructions.
          </p>
          <pre className="flex items-center justify-between rounded-sm bg-neutral-150 p-3 font-mono text-neutral-400">
            {type === 'LOCAL_DEMO' ? (
              <>
                <span>
                  <span className="select-none">$ </span>qovery demo up
                </span>
                <CopyButton content="qovery demo up" />
              </>
            ) : (
              <>
                <span>
                  <span className="select-none">$ </span>qovery cluster install
                </span>
                <CopyButton content="qovery cluster install" />
              </>
            )}
          </pre>
        </li>
        <li className="rounded border border-neutral-200 p-3">
          <h5 className="mb-1 text-sm font-medium">3. Deploy your first environment!</h5>
          <p className="font-normal text-neutral-350">
            Once the installation is completed, get back to the Qovery console and deploy your first environment on your
            brand new cluster.
          </p>
        </li>
      </ul>
    </>
  )
}

export default ClusterSetup
