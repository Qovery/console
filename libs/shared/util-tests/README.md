# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
