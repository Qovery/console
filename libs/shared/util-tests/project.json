{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
