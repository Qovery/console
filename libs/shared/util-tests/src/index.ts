export * from './lib/shared-util-tests'
