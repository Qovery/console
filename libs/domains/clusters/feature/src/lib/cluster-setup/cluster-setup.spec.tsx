import { renderWithProviders } from '@qovery/shared/util-tests'
import { ClusterSetup } from './cluster-setup'

describe('ClusterSetup', () => {
  it('should match manage deployment snapshot', async () => {
    const { baseElement } = renderWithProviders(<ClusterSetup type="SELF_MANAGED" />)
    expect(baseElement).toMatchSnapshot()
  })
})
