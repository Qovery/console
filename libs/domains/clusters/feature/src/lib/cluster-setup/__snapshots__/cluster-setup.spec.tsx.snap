// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterSetup should match manage deployment snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-row gap-x-3 p-3 border rounded text-sm border-sky-500 bg-sky-50 text-sky-500"
      data-accent-color="sky"
    >
      <div
        class="-order-2 text-lg leading-6"
      >
        <i
          aria-hidden="true"
          class="fa-regular fa-info-circle "
        />
      </div>
      <div
        class="-order-1 mr-auto gap-x-3 text-neutral-400"
      >
        <p>
          This guide walks you through the process of installing Qovery on your existing Kubernetes cluster, where you will have full control over the infrastructure. You will be responsible for managing updates and upgrades. Please note that advanced Kubernetes knowledge is required.
        </p>
      </div>
    </div>
    <ul
      class="flex flex-col gap-4 text-sm font-medium text-neutral-400"
    >
      <li
        class="rounded border border-neutral-200 p-3"
      >
        <h5
          class="mb-1 text-sm font-medium"
        >
          1. Download/Update Qovery CLI
        </h5>
        <p
          class="mb-2 font-normal text-neutral-350"
        >
          Download and install the Qovery CLI (or update its version to the latest version).
        </p>
        <a
          class="cursor-pointer transition duration-100 font-medium inline-flex items-center gap-1 text-sky-500 hover:text-sky-600 text-sm"
          href="https://hub.qovery.com/docs/using-qovery/interface/cli/#install"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://hub.qovery.com/docs/using-qovery/interface/cli/#install
          <i
            aria-hidden="true"
            class="fa-solid fa-arrow-up-right-from-square text-xs leading-5"
          />
        </a>
      </li>
      <li
        class="rounded border border-neutral-200 p-3"
      >
        <h5
          class="mb-1 text-sm font-medium"
        >
          2. 
          Install Qovery on your cluster
        </h5>
        <p
          class="mb-2  font-normal text-neutral-350"
        >
          Run the following command from your terminal and follow the instructions.
        </p>
        <pre
          class="flex items-center justify-between rounded-sm bg-neutral-150 p-3 font-mono text-neutral-400"
        >
          <span>
            <span
              class="select-none"
            >
              $ 
            </span>
            qovery cluster install
          </span>
          <button
            class="inline-flex items-center shrink-0 transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-xs h-7 px-2 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400 cursor-pointer gap-2 font-sans font-medium"
            type="button"
          >
            Copy
            <i
              aria-hidden="true"
              class="fa-solid fa-copy text-xs"
            />
          </button>
        </pre>
      </li>
      <li
        class="rounded border border-neutral-200 p-3"
      >
        <h5
          class="mb-1 text-sm font-medium"
        >
          3. Deploy your first environment!
        </h5>
        <p
          class="font-normal text-neutral-350"
        >
          Once the installation is completed, get back to the Qovery console and deploy your first environment on your brand new cluster.
        </p>
      </li>
    </ul>
  </div>
</body>
`;
